packages/webamp/js/components/Vis.tsx

```diff
@@ -0,0 +1,234 @@
+import React, {
+  useMemo,
+  useCallback,
+  useState,
+  useLayoutEffect,
+  useEffect,
+} from "react";
+
+import * as Actions from "../actionCreators";
+import * as Selectors from "../selectors";
+import { FFT } from "./FFTNullsoft";
+import { useTypedSelector, useActionCreator } from "../hooks";
+// import { usePaintOscilloscopeFrame } from "./useOscilloscopeVisualizer";
+// import { usePaintBarFrame, usePaintBar } from "./useBarVisualizer";
+import { VISUALIZERS, MEDIA_STATUS } from "../constants";
+
+import {
+  Vis as IVis,
+  VisPaintHandler,
+  BarPaintHandler,
+  WavePaintHandler,
+  NoVisualizerHandler,
+} from "./VisPainter";
+
+type Props = {
+  analyser: AnalyserNode;
+};
+
+export let PIXEL_DENSITY = 1;
+
+const fft = new FFT();
```

Comment:
Is it safe to have multiple instances of the visualizer (potentially in multiple instances of Webamp on the same page) share the same `FFT` object?

I feel like it would be safer to construct the instance as part of the component lifecycle maybe?